homeassistant/components/weather/__init__.py

```diff
@@ -110,6 +118,10 @@ def state_attributes(self):
         if attribution is not None:
             data[ATTR_WEATHER_ATTRIBUTION] = attribution
 
+        forecast = self.forecast
+        if forecast is not None:
+            data[ATTR_FORECAST] = forecast
```

Comment:
You will have to convert the units used in the forecast to the units of the user.